src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -486,14 +470,16 @@ public ClusterManager getClusterManager() {
 
   private void closeClusterManager(Handler<AsyncResult<Void>> completionHandler) {
```

Comment:
it would be more clean here to have `closeClusterManager` to consume a `Promise<Void>` so that the implementation can simply call complete() or fail() and not require a `runOnContext` operation

Reply:
I can do that, but I haven't actually touched these parts. How about futurizing the whole close process?